doc/whats_new/v0.23.rst

```diff
@@ -483,3 +520,15 @@ Miscellaneous
   `~sklearn.base.ClassifierMixin`. This tag is used to ensure that a proper
   error message is raised when y was expected but None was passed.
   :pr:`16622` by `Nicolas Hug`_.
+
+- |API| Most constructor and function parameters are now expected to be passed
```

Comment:
I was wondering whether we should have a top-level note for this one? Like we did in 0.22 for the [deprecations](https://scikit-learn.org/stable/whats_new/v0.22.html#clear-definition-of-the-public-api)

Reply:
yeah I was thinking about having a dedicated deprecation section too.

Reply:
Wanna do it? I'm happy to help if needed

Reply:
it'd be really nice if you did this part.